ui/app/models/task.js

```diff
@@ -58,9 +58,12 @@ export default class Task extends Fragment {
   @fragmentArray('volume-mount', { defaultValue: () => [] }) volumeMounts;
 
   async _fetchParentJob() {
-    let job = await this.store.findRecord('job', this.taskGroup.job.id, {
-      reload: true,
-    });
+    let job = await this.store.peekRecord('job', this.taskGroup.job.id);
+    if (!job) {
```

Comment:
Since task is a model fragment of a task group which is a model fragment of a job, I don't believe it's possible to ever have the task in the store and not the job.

Doesn't hurt to be defensive here I suppose, but I figured it was worth mentioning.